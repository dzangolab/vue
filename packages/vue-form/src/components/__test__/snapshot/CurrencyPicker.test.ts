import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import CurrencyPicker from "../../CurrencyPicker.vue";

import type { CurrencyOption } from "../../../types";

describe("CurrencyPicker", () => {
    const options = [
        { code: "AUD", label: "Australian Dollar", symbol: "$", value: "AUD" },
        { code: "GBP", label: "British Pound", symbol: "£", value: "GBP" },
        { code: "EUR", disabled: true, label: "Euro", symbol: "€", value: "EUR" },
        { code: "JPY", label: "Japanese Yen", symbol: "¥", value: "JPY" },
        { code: "USD", label: "US Dollar", symbol: "$", value: "USD" },
    ] as CurrencyOption[];

    it("matches snapshot", async () => {
        const wrapper = mount(CurrencyPicker, {
            props: {
                label: "Currency Picker",
                modelValue: ["AUD", "GBP"],
                multiple: true,
                name: "currency-picker",
                options: options,
                placeholder: "Select a currency",
            },
        });

    expect(wrapper.element).toMatchSnapshot();
  });
});
