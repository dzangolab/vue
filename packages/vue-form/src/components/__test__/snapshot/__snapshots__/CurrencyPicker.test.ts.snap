// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CurrencyPicker > matches snapshot 1`] = `
<div
  class="currency-selector"
>
  <div
    class="field currency-picker"
  >
    <label
      for="currency-picker"
    >
      Currency Picker
    </label>
    
    <div
      class="multiple-mode multiselect"
      id="input-field-currency-picker"
      name="currency-picker"
      tabindex="0"
      value="AUD,GBP"
    >
      <!--v-if-->
      <div
        class="multiselect-input"
      >
        <span
          class="multiselect-placeholder"
        >
          Select a currency
        </span>
        <span
          class="action-items"
        >
          <!--v-if-->
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Complete"
            >
              <g
                id="F-Chevron"
              >
                <polyline
                  fill="none"
                  id="down"
                  points="5 8.5 12 15.5 19 8.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </g>
            </g>
          </svg>
        </span>
      </div>
      <!--v-if-->
    </div>
    <!---->
    
  </div>
</div>
`;
